front/src/monthview/Monthview.js

```diff
@@ -0,0 +1,367 @@
+import React, { Component } from 'react';
+
+import '../App.css';
+import './Monthview.css';
+
+// Material UI components
+import { Link } from 'react-router-dom';
+
+// Moment for date management
+import moment from 'moment';
+import { viewChanger, getSchedulingFreeform, checkColor } from '../utils/Utils';
+
+// Translation
+import texts from '../texts/texts.json';
+
+const { weekdayShorthand } = texts;  // eslint-disable-line
+
+class Monthview extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      yearNro: 0,
+      monthNro: 0,
+    };
+
+    this.previousMonthClick = this.previousMonthClick.bind(this);
+    this.nextMonthClick = this.nextMonthClick.bind(this);
+    this.update = this.update.bind(this);
+  }
+
+  componentDidMount() {
+    this.update();
+    this.getYear();
+    this.getMonth();
+  }
+
+  componentWillReceiveProps() { // eslint-disable-line
+    this.update();
+  }
+
+  getYear = () => {
+    try {
+      const fullUrl = window.location.href.split('/');
+      const urlParamDate = fullUrl[5];
+
+      const urlParamDateSplit = urlParamDate.split('-');
+
+      const paramYear = urlParamDateSplit[0];
+      if (paramYear.length === 4) {
+        this.setState({ yearNro: paramYear });
+        return paramYear;
+      }
+    } catch {
+      console.log('Error getting year from url');
+    }
+    const today = new Date();
+    const yyyy = today.getFullYear();
+    this.setState({ yearNro: yyyy });
+    return yyyy;
+  }
+
+  getMonth = () => {
+    try {
+      const fullUrl = window.location.href.split('/');
+      const urlParamDate = fullUrl[5];
+
+      const urlParamDateSplit = urlParamDate.split('-');
+
+      const paramMonth = urlParamDateSplit[1].padStart(2, '0');
+      if (paramMonth.length === 2) {
+        this.setState({ monthNro: paramMonth });
+        return paramMonth;
+      }
+    } catch {
+      console.log('Error getting month from url');
+    }
+    const today = new Date();
+    const mm = String(today.getMonth() + 1).padStart(2, '0');
+    this.setState({ monthNro: mm });
+    return mm;
+  }
+
+  createWeekDay = () => {
+    const table = [];
+    let dayNumber;
+
+    const fin = localStorage.getItem('language'); // eslint-disable-line
```

Comment:
No need to disable eslint here and row 89

Reply:
Fixed